osbrain/agent.py

```diff
@@ -209,6 +210,9 @@ def __init__(self, name=None, host=None, serializer=None, transport=None,
                  attributes=None):
         self.uuid = unique_identifier()
         self.name = name
+        if not self.name:
+            self.name = self.uuid.decode()
+
```

Comment:
It seems I cannot review parts of the code that were not changed... :sweat_smile: 

Anyway... Here you will probably want to move the `if attributes:` section to a new method `_set_attributes(self, attributes):`.

Reply:
Why exactly? Is it just a matter of style?

Reply:
`__init__` became too complex, which usually tells us that we should probably split this method/function in smaller (less complex) parts. This eases the process of testing and code reusing.

I suggested the `if attributes` section as it seems it has its own meaning and adds 3 to the cyclomatic complexity. So that is a perfect candidate for a split.

You can even add it as a function rather than a method. In example: `def add_dictionary_attributes(target, attributes):`. As this could be applied to any object, not just `Agent`.

Reply:
PS: this refactoring (i.e.: moving that section into a new function) should be a separate commit. Keep it in this PR though.

Reply:
PSS: :joy: Even if that section is covered, you might want to add simple tests to verify the new function.

Reply:
OK, then. I agree that is a separate issue and it should be kept in a separate commit.

Reply:
Also, I just tested it out and now the tests seem to pass 👍 .